domain/src/test/java/org/oppia/android/domain/topic/TopicListControllerTest.kt

```diff
@@ -228,47 +230,54 @@ class TopicListControllerTest {
   }
 
   @Test
-  fun testRetrieveOngoingStoryList_defaultLesson_hasCorrectInfo() {
-    topicListController.getOngoingStoryList(profileId0).toLiveData()
-      .observeForever(mockOngoingStoryListObserver)
+  fun testRetrievePromotedActivityList_defaultLesson_hasCorrectInfo() {
+    topicListController.getPromotedActivityList(profileId0).toLiveData()
+      .observeForever(mockPromotedActivityListObserver)
     testCoroutineDispatchers.runCurrent()
 
-    verifyGetOngoingStoryListSucceeded()
-    verifyDefaultOngoingStoryListSucceeded()
+    verifygetPromotedActivityListSucceeded()
   }
 
   @Test
-  fun testRetrieveOngoingStoryList_markRecentlyPlayedFracStory0Exp0_ongoingStoryListIsCorrect() {
+  fun testGetPromotedActivityList_markRecentlyPlayedFracStory0Exp0_promotedActivityListIsCorrect() {
     storyProgressController.recordRecentlyPlayedChapter(
       profileId0,
       FRACTIONS_TOPIC_ID,
       FRACTIONS_STORY_ID_0,
       FRACTIONS_EXPLORATION_ID_0,
       getCurrentTimestamp()
     )
+    testCoroutineDispatchers.runCurrent()
 
-    val ongoingTopicList = retrieveOngoingStoryList()
-    assertThat(ongoingTopicList.recentStoryCount).isEqualTo(1)
-    verifyOngoingStoryAsFractionStory0Exploration0(ongoingTopicList.recentStoryList[0])
+    val promotedActivityList = retrievePromotedActivityList()
+    with(promotedActivityList.promotedStoryList) {
+      assertThat(recentlyPlayedStoryCount).isEqualTo(1)
+      verifyPromotedStoryAsFractionStory0Exploration0(recentlyPlayedStoryList[0])
+      assertThat(suggestedStoryCount).isEqualTo(1)
+      verifyOngoingStoryAsRatioStory0Exploration0(suggestedStoryList[0])
```

Comment:
This is slightly confusing to me--is it actually an ongoing story that's being verified here?

Reply:
I would suggest not to use `with` here as it makes difficult to understand different assertions.

Reply:
Ongoing story and also it will suggest next topic.

Reply:
I also agree with @aggarwalpulkit596 here. I was thrown off by the 'with' and it does muddle things a bit.

Reply:
Done.